src/Service/Terminal/Service_Terminal.re

```diff
@@ -156,6 +159,11 @@ module Sub = {
           ReveryTerminal.resize(~rows, ~columns, state.terminal);
           state.isResizing := false;
           {...state, rows, columns};
+        } else if (state.queuedScreen^ != None) {
+          let screen = Option.get(state.queuedScreen^);
+          dispatch(ScreenUpdated({id: params.id, screen}));
+          state.queuedScreen := None;
+          state;
```

Comment:
This seems to put a weird dependency between the state being updated and the terminal updating, which can lead to some unintuitive bugs. If we manage to reduce the number of status updates, which is a good thing, the terminal might start to feel unresponsive. And if the number of state updates increase, in the same manner as this would if dispatched directly, it still won't solve the problem but actually exacerbate it.

I'm not sure what downside there is to just using a timer, other than needing a bit more wiring, but it would make the system as a whole significantly simpler I think.

Reply:
Good point - I moved to a timer / debounce strategy, which should be more reliable than relying on the timing of the `update`s.